File: front/src/app/shared/model/dto/pessoa-dto.ts
// pessoa.dto.ts

import {StatusEnum} from "@/shared/model/enum/status.enum";
import {ClienteDto} from "@/shared/model/dto/cliente-dto";

export interface PessoaDto {
    id?: number;
    cliente?: ClienteDto;
    nome?: string;
    cpf?: string;
    telefone?: string;
    email?: string;
    status?: StatusEnum;
    endereco?: string;
    complemento?: string;
    cidade?: string;
    estado?: string;
    cep?: string;
    senha?: string;
    criadoEm?: Date;
    alteradoEm?: Date;
    version?: number;
}

File: back/src/main/java/br/eng/eliseu/presente/model/dto/PessoaDto.java
package br.eng.eliseu.presente.model.dto;

import br.eng.eliseu.presente.model.StatusEnum;

import java.time.LocalDateTime;

public record PessoaDto(
        Long id,

        ClienteDto cliente,

        String nome,
        String cpf,
        String telefone,
        String email,

        StatusEnum status,

        String endereco,
        String complemento,
        String cidade,
        String estado,
        String cep,

        String senha,

        LocalDateTime criadoEm,
        LocalDateTime alteradoEm,

        Long version

) {}